llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp

```diff
@@ -1237,10 +1237,19 @@ translateSpirvGlobalUses(LoadInst *LI, StringRef SpirvGlobalName,
     NewInst = llvm::Constant::getNullValue(LI->getType());
   } else if (isa<GetElementPtrConstantExpr>(LI->getPointerOperand())) {
     // Translate the load that has getelementptr as an operand
-    auto *GEPCE = cast<GetElementPtrConstantExpr>(LI->getPointerOperand());
-    uint64_t IndexValue =
-        cast<Constant>(GEPCE->getOperand(2))->getUniqueInteger().getZExtValue();
-    NewInst = generateSpirvGlobalGenX(LI, SpirvGlobalName, IndexValue);
+    auto *GEPOp = cast<GEPOperator>(LI->getPointerOperand());
```

Comment:
You should be able to replace this with `} else if (auto *GEPOp = cast<GEPOperator>(LI->getPointerOperand())) {`

Reply:
fixed thx